fe/lib/components/JobCategorySelect/JobCategorySelectInput.tsx

```diff
@@ -0,0 +1,96 @@
+import { Column, Columns, Dropdown, Stack } from 'braid-design-system';
+import React, { useEffect, useState } from 'react';
+
+import { JobCategory } from '../../types/seek.graphql';
+import { findCategory } from '../../utils';
+
+interface Props {
+  jobCategories: JobCategory[];
+  onSelect: (jobCategory: JobCategory) => void;
+}
+
+const JobCategorySelectInput = ({
+  jobCategories,
+  onSelect,
+  ...restProps
+}: Props) => {
+  const [selectedParentCategoryId, setSelectedParentCategoryId] = useState('');
+  const [selectedChildCategoryId, setSelectedChildCategoryId] = useState('');
+  const [childCategories, setChildCategories] = useState<JobCategory[]>();
+
+  useEffect(() => {
+    if (selectedParentCategoryId !== '') {
+      const parentCategory = findCategory(
+        jobCategories,
+        selectedParentCategoryId,
+      );
+
+      if (parentCategory?.children) {
+        setChildCategories(parentCategory.children);
+        setSelectedChildCategoryId('');
+        onSelect(parentCategory);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Just for my benefit, which deps are problematically changing between renders here?

Reply:
It had re-rendering issuers with `setChildCategories`, `setSelectedChildCategoryId` and `onSelect`. It was causing extraneous renders. We could look into using useCallback in the future to see if they can be mitigated.